modules/fluid_properties/test/src/userobjects/NaNInterfaceTestFluidProperties.C

```diff
@@ -0,0 +1,37 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NaNInterfaceTestFluidProperties.h"
+
+registerMooseObject("FluidPropertiesTestApp", NaNInterfaceTestFluidProperties);
+
+template <>
+InputParameters
+validParams<NaNInterfaceTestFluidProperties>()
+{
+  InputParameters params = validParams<SinglePhaseFluidProperties>();
+  params += validParams<NaNInterface>();
+
+  params.addClassDescription("Fluid properties for testing NaNInterface");
+
+  return params;
+}
+
+NaNInterfaceTestFluidProperties::NaNInterfaceTestFluidProperties(const InputParameters & parameters)
+  : SinglePhaseFluidProperties(parameters), NaNInterface(this)
+{
+}
+
+NaNInterfaceTestFluidProperties::~NaNInterfaceTestFluidProperties() {}
```

Comment:
If you aren't defining a new base class just leave out the empty destructor. We only need to define these when we want to declare them as virtual. If you do need to define them for some reason (rare), you can just use "= default;"